hack/tools/golangci-lint/sortedfeatures/sortedfeatures.go

```diff
@@ -0,0 +1,230 @@
+	filename := pass.Fset.File(pass.Files[0].Pos()).Name()
+	isTargetFile := false
+
+	// Determine which files to check
+	var targetFiles []string
+	if len(config.Files) > 0 {
+		// If specific files are provided, only check those
+		targetFiles = config.Files
+	} else {
+		// Otherwise use the default target files
+		targetFiles = defaultTargetFiles
+	}
+
+	if config.Debug {
+		fmt.Printf("Checking file: %s\n", filename)
+	}
+
+	for _, target := range targetFiles {
+		if strings.HasSuffix(filename, target) || strings.HasSuffix(filename, filepath.Base(target)) {
+			isTargetFile = true
+			break
+		}
+	}
+
+	if !isTargetFile {
+		return nil, nil
+	}
+
+	for _, file := range pass.Files {
+		ast.Inspect(file, func(n ast.Node) bool {
+			switch decl := n.(type) {
+			case *ast.GenDecl:
+				if decl.Tok == token.CONST || decl.Tok == token.VAR {
+					checkSorting(pass, decl)
+				}
+			}
+			return true
+		})
+	}
+	return nil, nil
+}
+
+// isFeatureGateDeclaration checks if a value spec is likely a feature gate declaration
+func isFeatureGateDeclaration(spec *ast.ValueSpec) bool {
+	// Check if there's a type with a selector expression that ends with Feature
+	if spec.Type != nil {
+		if selectorExpr, ok := spec.Type.(*ast.SelectorExpr); ok {
+			return selectorExpr.Sel.Name == "Feature"
+		}
+	}
+
+	// Check if there's a value that's a string literal matching the name pattern
+	if len(spec.Values) > 0 {
+		if basicLit, ok := spec.Values[0].(*ast.BasicLit); ok && basicLit.Kind == token.STRING {
+			// Feature gates often have their name as the string value
+			for _, name := range spec.Names {
+				if strings.Contains(basicLit.Value, name.Name) {
+					return true
+				}
+			}
+		}
+
+		// Check for framework.WithFeature pattern used in test/e2e/feature/feature.go
+		if callExpr, ok := spec.Values[0].(*ast.CallExpr); ok {
+			if selExpr, ok := callExpr.Fun.(*ast.SelectorExpr); ok {
+				if selExpr.Sel.Name == "WithFeature" || selExpr.Sel.Name == "Feature" {
+					return true
+				}
+			}
+		}
+	}
+
+	return false
+}
+
+func checkSorting(pass *analysis.Pass, decl *ast.GenDecl) {
+	if len(decl.Specs) <= 1 {
+		return // Nothing to sort with just one item
+	}
+
+	// Extract feature names and their positions
+	type featureInfo struct {
+		Name string
+	}
+
+	var features []featureInfo
+	var hasFeatureGates bool
+
+	for _, spec := range decl.Specs {
+		if valueSpec, ok := spec.(*ast.ValueSpec); ok {
+			// Only check specs that look like feature gates
+			if isFeatureGateDeclaration(valueSpec) {
+				hasFeatureGates = true
+				for _, name := range valueSpec.Names {
+					features = append(features, featureInfo{
+						Name: name.Name,
+					})
+				}
+			}
+		}
+	}
+
+	// If we didn't find any feature gates, don't check sorting
+	if !hasFeatureGates || len(features) <= 1 {
+		return
+	}
+
+	// Check if the features are sorted
+	isSorted := true
+	for i := 1; i < len(features); i++ {
+		if strings.Compare(features[i-1].Name, features[i].Name) > 0 {
```

Comment:
This is sorting in pure ASCII order, so `CPUCFSQuotaPeriod` sorts before `ClusterTrustBundle`... I think it would be better to do `strings.Compare(strings.ToLower(features[i-1].Name, strings.ToLower(features[i].Name))` ?

Reply:
@danwinship I think i key-ed off this comment @danwinship 

https://cs.k8s.io/?q=Feature%20gates%20should%20be%20listed%20in%20alphabetical%2C%20case-sensitive&i=nope&literal=nope&files=&excludeFiles=&repos=kubernetes/kubernetes

https://github.com/kubernetes/kubernetes/blob/ccf291b50166ce82204486aae6a626a30b469182/pkg/features/kube_features.go#L38-L41

Reply:
I've updated all the files to have similar header so it is very clear.